reactive/coredns.py

```diff
@@ -47,3 +81,34 @@ def start_charm():
 def update_image():
     # handle a new image resource becoming available
     clear_flag('charm.coredns.started')
+
+
+@when('charm.coredns.started', 'coredns.joined')
+def send_ip():
+    """
+    Send CoreDNS IP to kuberentes-worker
+    """
+    try:
+        coredns = endpoint_from_flag('coredns.joined')
+        if coredns:
+            service_ip = get_service_ip('coredns')
+            if service_ip:
+                coredns.send_ip(service_ip)
+                clear_flag('coredns.joined')
+    except Exception as e:
+        hookenv.log("Failed sending CoreDNS IP: {}".format(e))
+
+
+def get_service_ip(endpoint):
+    try:
+        info = hookenv.network_get(endpoint, hookenv.relation_id())
+        if 'ingress-addresses' in info:
+            addr = info['ingress-addresses'][0]
+            if len(addr):
```

Comment:
You log no ingress addresses, but don't log about a lack of addr. Should that be logged as well?